Framework/Crystal/src/PredictPeaks.cpp

```diff
@@ -268,9 +287,14 @@ void PredictPeaks::exec() {
   Progress prog(this, 0.0, 1.0, possibleHKLs.size() * gonioVec.size());
   prog.setNotifyStep(0.01);
 
-  m_detectorCacheSearch = std::make_unique<DetectorSearcher>(m_inst, m_pw->detectorInfo());
+  if (!(m_leanElasticPeak && !leanElasticPeak_calculate_wl))
+    m_detectorCacheSearch = std::make_unique<DetectorSearcher>(m_inst, m_pw->detectorInfo());
 
-  if (getProperty("CalculateGoniometerForCW")) {
+  if (m_leanElasticPeak && !leanElasticPeak_calculate_wl) {
+    for (auto &possibleHKL : possibleHKLs) {
+      calculateQAndAddToOutputLeanElastic(possibleHKL, ub, gonioVec.front());
```

Comment:
In this case, we don't just want to loop over every possible goniometer because you will just end up with `gonioVec.size()` copies of the peaks just with different goniometers. I think I will change this to just use the identity matrix and make it clear in the documentation that is what's happening.